internal/ad/backends/winbind/winbind_test.go

```diff
@@ -63,19 +71,62 @@ func TestWinbind(t *testing.T) {
 				config.ADServer = tc.staticADServer
 			}
 
-			backend, err := winbind.New(context.Background(), config)
+			kinitCmdOutputFile := filepath.Join(t.TempDir(), "kinit-output")
+			kinitCmd := []string{"env", "GO_WANT_HELPER_PROCESS=1", os.Args[0], "-test.run=TestExecuteKinitCommand", "--", kinitCmdOutputFile}
+			if tc.wantKinitErr {
+				kinitCmd = append(kinitCmd, "-Exit1-")
+			}
+
+			backend, err := winbind.New(context.Background(), config, hostname, winbind.WithKinitCmd(kinitCmd))
 			if tc.wantErr {
 				require.Error(t, err, "New should have errored out")
 				return
 			}
 
 			got := testutils.FormatBackendCalls(t, backend)
+
+			// Check kinit command
+			gotKinitCmd := []byte("not executed\n")
```

Comment:
We thus have "not executed" in the test golden file written in case of error requested in kinit. When reading it, I feel this to be a little bit weird… (also, it doesn’t differentiate not being executed and erroring out).